### PR TITLE
Mock Github Server to avoid rate limiting in HTTP Client Guide

### PR DESCRIPTION
**Pull request by sdelamo**
This PR updates to 3.8.4

### CODE REVIEW

File: guides/micronaut-http-client/groovy/src/test/groovy/example/micronaut/GithubControllerSpec.groovy [hunk 1]
```diff
@@ -1,60 +1,77 @@
...
+        }
 
-        then:
-        githubReleases.stream()
-                .map(GithubRelease::getName)
-                .allMatch(name -> MICRONAUT_RELEASE.matcher(name)
-                        .find())
+        @Consumes(value = ["application/vnd.github.v3+json", "application/json"])
+        @Get("/repos/micronaut-projects/micronaut-core/releases") // <3>
+        String coreReleases() {
```
Comment (sdelamo) [inline guides/micronaut-http-client/groovy/src/test/groovy/example/micronaut/GithubControllerSpec.groovy:71]: I don't know if it is Apache Groovy or a bug in Micronaut Framework. If I set the same as in Kotlin or Java (see below), it does not work. 

```groovy
Optional<InputStream> coreReleases() {
            resourceLoader.getResourceAsStream("releases.json")
}

Comment (melix) [reply guides/micronaut-http-client/groovy/src/test/groovy/example/micronaut/GithubControllerSpec.groovy:71]: I've seen cases where Groovy didn't resolve resources with the same path. You could try to use an absolute path to the resource instead.

**Review by melix — APPROVED** [review]

File: guides/micronaut-http-client/groovy/src/test/groovy/example/micronaut/GithubControllerSpec.groovy (hunk 1 above)
Comment (sdelamo) [reply guides/micronaut-http-client/groovy/src/test/groovy/example/micronaut/GithubControllerSpec.groovy:71]: Please, note that `resourceLoader.getResourceAsStream("releases.json")` seems to work. In order to make it work, in Apache Groovy I have to unwrap the optional and get the string of input stream with Groovy's `.text`. Thus, I think it is not an issue with the path. It seems the issue is returning an `Optional<InputStream>` with Apache Groovy.

**Review by timyates — APPROVED** [review]

File: guides/micronaut-http-client/groovy/src/test/groovy/example/micronaut/GithubControllerSpec.groovy (hunk 1 above)
Comment (melix) [reply guides/micronaut-http-client/groovy/src/test/groovy/example/micronaut/GithubControllerSpec.groovy:71]: Ah sorry I missed it wasn't a `ClassLoader` here but a resource loader.

**Pull request merged**